--- libcurl-7.21.7.old/src/Makefile.in	2011-12-21 00:49:48.825354179 +0200
+++ libcurl-7.21.7/src/Makefile.in	2011-12-21 00:50:26.037368519 +0200
@@ -455,7 +455,7 @@
        tool_xattr.h

 curl_SOURCES = $(CURL_CFILES) $(CURLX_ONES) $(CURL_HFILES)
-curl_LDADD = $(top_builddir)/lib/libcurl.la @CURL_LIBS@ @LIBMETALINK_LIBS@
+curl_LDADD = $(top_builddir)/lib/libcurl.la @LIBCURL_LIBS@ @LIBMETALINK_LIBS@
 curl_LDFLAGS = @LIBMETALINK_LDFLAGS@
 curl_DEPENDENCIES = $(top_builddir)/lib/libcurl.la
 BUILT_SOURCES = hugehelp.c
